algorithms/arrays/flatten.py

```diff
@@ -3,29 +3,27 @@
 Given an array that may contain nested arrays,
 produce a single resultant array.
 """
-from collections.abc import Iterable
-
 
 # return list
 def flatten(input_arr, output_arr=None):
-    if output_arr is None:
-        output_arr = []
-    for ele in input_arr:
-        if not isinstance(ele, str) and isinstance(ele, Iterable):
-            flatten(ele, output_arr)    #tail-recursion
-        else:
-            output_arr.append(ele)      #produce the result
-    return output_arr
+	if output_arr is None:
+		output_arr = []
+	for ele in input_arr:
+		try:
+			if len(ele) > 0:
```

Comment:
Not every object that has `__len__` implemented is `iterable`. I don't see how this improves the existing method.